src/posix/platform/trel_udp6.cpp

```diff
@@ -420,7 +445,13 @@ void otPlatTrelUdp6Init(otInstance *aInstance, const otIp6Address *aUnicastAddre
     sockAddr.sin6_port     = htons(sUdpPort);
     sockAddr.sin6_addr     = in6addr_any;
     sockAddr.sin6_scope_id = (uint32_t)sInterfaceIndex;
-    VerifyOrDie(bind(sMulticastSocket, (struct sockaddr *)&sockAddr, sizeof(sockAddr)) != -1, OT_EXIT_ERROR_ERRNO);
+
+    if (bind(sMulticastSocket, (struct sockaddr *)&sockAddr, sizeof(sockAddr)) == -1)
```

Comment:
```suggestion
    if (bind(sMulticastSocket, reinterpret_cast<struct sockaddr *>(&sockAddr), sizeof(sockAddr)) == -1)
```

Reply:
Good point! I think this file was originally a `c` file so it uses c casting (and c style) all over.
I'd suggest we consider changing this in a follow-up/next PR (and keep this PR to adding new logs only).